File: falatu-backend/src/http/chat/datasources/chat.datasource.ts
import { ResultWrapper } from "src/utils/result/ResultWrapper";
import { CreateChatDTO } from "../dtos/create_chat.dto";
import { ChatEntity } from "../entities/chat.entity";

export abstract class ChatDatasource {
  abstract createChat(params: CreateChatDTO): Promise<ResultWrapper<ChatEntity>>;
}

File: frontend/falatu_mobile/lib/chat/core/data/datasources/chat/chat_datasource.dart
import "package:falatu_mobile/chat/core/data/models/chat/create_chat_model.dart";
import "package:falatu_mobile/chat/core/domain/entities/chat/chat_entity.dart";
import "package:falatu_mobile/commons/core/domain/entities/result_wrapper.dart";

abstract class ChatDatasource {
  Future<ResultWrapper<Stream<List<ChatEntity>>>> loadChats();

  ResultWrapper<Stream<ChatEntity>> createChat(CreateChatModel params);

}
